src/modules/commands/expAdd.js

```diff
@@ -0,0 +1,46 @@
+const isEmpty = require('lodash/isEmpty');
+
```

Comment:
в идеале ВЕСЬ ЭТОТ МОДУЛЬ должен выглядеть как модуль в котором ровно одна строчка кода
```suggestion
module.exports = [isModerator, command('expAdd target amount reason'), addExp]
```

где addExp это универсальный модуль, который смотрит args на наличие указанных выше параметров, которые любой другой модуль может сетапить

Reply:
Т.е. как это было сделано раньше...*see_no_evil*

Reply:
Нет, раньше оно не смотрело request а просто каррировало количество опыта. Раньше оно занималось хуйней и просто обновляло некий request.exp. Сейчас должно сразу же обновлять опыт у юзера и смотреть аргументы из request (чтобы тот же command мог бы прозрачно передавать данные)

Reply:
На самом деле, почему в args добавлять -- тоже не понятно, это ж для команд
В смысле сразу же добавлять, если "любой другой модуль может сетапить" и мы должны "смотреть аргументы из request"?
Как было: писалось хоть откуда `req.exp = someAmount`, в отдельном модуле обновлялось значение в бд

Reply:
Модуль - самодостаточная единица, оптимизации на уровне, когда ты ещё не понимаешь что оптимизируешь - всратая идея

Reply:
Насчёт args спорно. Но суть такая что один модуль устанавливает значения, другой их принимает и юзает. Для кейса - команда даёт опыт, юзать args норм. Если ты создашь хранилище в request отдельное, то у тебя будет модуль который из args просто перекладывает в другое место (object mapping) что тоже норм.

Reply:
Ну, окей, т.е. вся разница со старым должна быть в том, чтобы вместо одного обновления делать обновления каждый раз
Оки)